Close #25. Close #34. Add bonus item functionality and sound effects.

ryanflach opened the pull request:
- Update obstacles so that motorcycles are now slightly slower than buildings and street dogs, as they would in reality also be driving, just not as fast as the jeepney. 
- Update obstacle placement so that random placement does not begin until the player has passed 1000 points. Prior to that point only street dogs appear on the screen.
- Add bonus items. Bonus items appear at 5 second intervals and are randomized between mango and lumpia. If at full health the player earns 500 bonus points. Otherwise, they regain one health.
- Add GitHub logo to bottom of index page that links to the repo.
- Add AudioPlayer class to control sound effects in jeepney, game, and obstacles classes.
- Add game sound effects
  - jeepney drive
  - jeepney jump
  - background audio (Filipino national anthem)
  - bonus collect
  - motorcycle hit
  - dog hit
- Add some additional tests for more robust coverage.
- Minor refactoring throughout. Remove outdated CSS.

ryanflach committed:
Update description of game

ryanflach committed:
Add GitHub logo link to repo

ryanflach committed:
Add bonus items, sound effects, and additional tests

allindow commented on lib/asset_manager.js:
```diff
@@ -164,6 +164,43 @@ AssetManager.prototype.randomBackgroundObject = function(previousItem, difficult
 
 // Bonus Objects
+AssetManager.prototype.bonusObject = function(itemName, difficulty) {
+  const lumpiaSpecs = {
```
lumpiaSpecs..best ever.

ryanflach replied:
My life as a developer is better having been able to have that as a legitimate variable name.

allindow commented:
Wowza @ryanflach this is some awesome work! LOVE the sound effects, LOVE the flying heart when you hit a bonus item. And I think it's super smart that the motorcycles don't show up until you go a little faster.

ryanflach committed:
Update location of GitHub link
- Update link to be a set distance from the canvas, rather than locked to the bottom. This prevents overlap of the logo with the canvas when the screen's height is shortened
